cmd/apps/openfaas_app.go

```diff
@@ -54,17 +54,17 @@ func MakeInstallOpenFaaS() *cobra.Command {
 	openfaas.Flags().StringArray("set", []string{}, "Use custom flags or override existing flags \n(example --set=gateway.replicas=2)")
 
 	openfaas.RunE = func(command *cobra.Command, args []string) error {
-		kubeConfigPath := config.GetDefaultKubeconfig()
 		wait, err := command.Flags().GetBool("wait")
 		if err != nil {
 			return err
 		}
-		if command.Flags().Changed("kubeconfig") {
-			kubeConfigPath, _ = command.Flags().GetString("kubeconfig")
-		}
 
+		kubeConfigPath, _ := command.Flags().GetString("kubeconfig")
+		if err := config.SetKubeconfig(kubeConfigPath); err != nil {
+			return err
+		}
 		fmt.Printf("Using kubeconfig: %s\n", kubeConfigPath)
-
+		os.Setenv("KUBECONFIG", kubeConfigPath)
```

Comment:
If `KUBECONFIG` is defined but the `--kubeconfig` flag is not set, this appears to overwrite `KUBECONFIG` with nothing causing a fallback to `$HOME/.kube/config`.